libs/librepcb/common/fileio/filepath.cpp

```diff
@@ -260,16 +260,16 @@ QString FilePath::cleanFileName(const QString& userInput, CleanFileNameOptions o
 {
     // perform compatibility decomposition (NFKD)
     QString ret = userInput.normalized(QString::NormalizationForm_KD);
+    // remove all invalid characters (note: backslashes are required to escape minus sign)
+    ret.remove(QRegularExpression("[^ _\\-.0-9A-Za-z]"));
```

Comment:
[Wikipedia says](https://en.wikipedia.org/wiki/Regular_expression#POSIX_basic_and_extended):
> Within POSIX bracket expressions, the dot character matches a literal dot. For example, a.c matches "abc", etc., but [a.c] matches only "a", ".", or "c".

I also found another interesting fact:
> The - character is treated as a literal character if it is the last or the first (after the ^, if present) character within the brackets: [abc-], [-abc]. Note that backslash escapes are not allowed.

If I understand correctly, backslash escapes are only allowed in [POSIX extended](https://en.wikipedia.org/wiki/Regular_expression#POSIX_extended) (which Qt seems to support, because the escaping works as expected).

But actually it may still be better to use following code:
```
ret.remove(QRegularExpression("[^-._ 0-9A-Za-z]"));
```